src/data_map.rs

```diff
@@ -151,3 +316,145 @@ impl Debug for ChunkInfo {
         )
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
not sure if this test reflects the current usage of the recursive datamap that used by autonomi client.
and whether the old recursive datamap can be backward compatible by this way, or the autonomi client shall carry out extra backward compatible work.